### DISCUSSION TITLE
Shall we add Maximum movable particles in HPMC?

### DISCUSSION THREAD

**Discussion by yda-cyber** [glotzerlab/hoomd-blue#1974] [General]
The HPMC currently seems only supporting the movement of all particles at each step.  This works very efficiently.

In a system that exist external field or has pair potentials, however, this has resulted that dU is scaling with N^2 (where N is the number of particle). 

As a result, the larger the system is, the higher the rejection ratio is (or, in other words, the meaning of temperature is not clear).

Shall we add a extra paramter of (Maximum movable particles per step) in HPMC? This presumably would correct the behavior.

**Comment by joaander** [comment]
The acceptance ratio of local trial moves (translation/rotation) does not depend on N (assuming all particles are in similar local environments). One time step performs many local trial moves which are accepted/rejected individually.

You can set `nselect` to control the number of trial moves attempted per particle per step (approximately), but cannot set it to a fractional value.
